lib/Targets/AIETargetXAIEV2.cpp

```diff
@@ -335,21 +335,27 @@ mlir::LogicalResult AIETranslateToXAIEV2(ModuleOp module, raw_ostream &output) {
       int acqValue = 0, relValue = 0;
       StringRef acqEnable = disable;
       StringRef relEnable = disable;
-      int lockID;
+      int acqLockID, relLockID;
       for (auto op : block.getOps<UseLockOp>()) {
         LockOp lock = dyn_cast<LockOp>(op.getLock().getDefiningOp());
-        lockID = lock.getLockIDValue();
         if (op.acquire() || op.acquire_ge()) {
           acqEnable = enable;
+          acqLockID = lock.getLockIDValue();
           acqValue = op.getLockValue();
+          if (op.acquire_ge())
+            acqValue = -acqValue;
         } else if (op.release()) {
           relEnable = enable;
+          relLockID = lock.getLockIDValue();
           relValue = op.getLockValue();
+
         } else {
           // unreachable for current targets
           return op.emitOpError("unsupported lock action");
         }
       }
+      // if((acqEnable == disable) || (relEnable == disable))
+      //   return memOp.emitOpError("XAIEV2 requires acquire and release");
```

Comment:
Remove or enable this? There are a few more instances of this comment block later in the file.

Reply:
Yeah, I think there's another bug there, was trying to figure out whether to deal with it in the same PR or not.